src/main/java/org/eclipse/yasson/internal/serializer/types/EnumSerializer.java

```diff
@@ -12,26 +12,57 @@
 
 package org.eclipse.yasson.internal.serializer.types;
 
+import java.util.EnumMap;
+
 import jakarta.json.stream.JsonGenerator;
 
 import org.eclipse.yasson.internal.SerializationContextImpl;
+import org.eclipse.yasson.internal.model.ClassModel;
+import org.eclipse.yasson.internal.model.PropertyModel;
+
 
 /**
  * Serializer of the {@link Enum} types.
  */
 class EnumSerializer extends TypeSerializer<Enum<?>> {
 
+    private final EnumMap<? extends Enum<?>, String> constantToNameMap;
+
     EnumSerializer(TypeSerializerBuilder serializerBuilder) {
         super(serializerBuilder);
+
+        constantToNameMap = createConstantToNameMap(serializerBuilder);
+    }
+
+    private static <E extends Enum<E>> EnumMap<E, String> createConstantToNameMap(TypeSerializerBuilder serializerBuilder) {
```

Comment:
Isn't this the same as in EnumDeserializer? Maybe it could be placed on some common place :-)

Reply:
I nearly agree with you, but there are two differences. First, we have a map String to Enum in one case and vice verse in another. And second: in one case we need a read name from the model and a write name in another. The rest dealing with enum is just the same.

So I can extract this and put in some common place with two parameters describing these differences. Or just one parameter for the name, and then just reverse the result map in one case or another.

The other question is: where is this common place? Because I see no common or util package here. I just had this refactoring today in the Enum tests. There I have the same code in adapter and de- and serializer. So I extracted the code from the adapter and use it then directly from de- and serializer. I left the common code in the adapter package and use it in the serializer, but I think it is OK to have this kind of import in the tests.